toolkit/scripts/kernel_sources_analysis.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for new configs which should be in required kernel configs json
+# Usage: python3 check_new_kernel_configs.py --required_configs <path to json of required configs> --config_str <string of diff for config file>
+
+import re
+
+def get_data_from_config(input_file):
+    with open(input_file, 'r') as file:
+        input_config_data = file.read()
+    return input_config_data
+
+def extract_kernel_dir_name(input_file):
+    match = re.search(r'SPECS/(.*?)/', input_file)
```

Comment:
>match = re.search(r'SPECS/(.*?)/', input_file) [](http://example.com/codeflow?start=4&length=46)

When I think about it now, using a regular expression here might be a bit of an overkill. Also, this now depends on the kernel spec being somewhere inside a directory called "SPECS", which makes this function more vulnerable to failures in case of changes in the future (a kernel spec placed under a different path).

If it is, however, required that all paths passed into this function contain the "SPECS" directory, then I agree this needs to be there. But in such case, I'd also add a check for that requirement.

Otherwise, I'd rely on a simple `os.path.basename(os.path.dirname(input_file))` (example: [here]([https://note.nkmk.me/en/python-os-basename-dirname-split-splitext/#:~:text=dirname()-,Use%20os.,name)%20from%20a%20path%20string.&text=If%20you%20want%20to%20get,above%20the%20file%2C%20use%20os](https://note.nkmk.me/en/python-os-basename-dirname-split-splitext/#:~:text=dirname()-,Use%20os.,name)%20from%20a%20path%20string.&text=If%20you%20want%20to%20get,above%20the%20file%2C%20use%20os).)).

Also, since the function will work (and I'm guessing is expected to work) for any file path, not only a path to a kernel spec, I think we can make the name more generic `extract_file_dir_name`.

Reply:
Replying just to try and force to make GitHub show this comment again.:)

Reply:
Done. I also added the optional argument "kernel" to pass the kernel name explicitly to the program